source/MRMesh/MRMeshTrimWithPlane.cpp

```diff
@@ -92,7 +93,15 @@ FaceBitSet subdivideWithPlane( Mesh & mesh, const Plane3f & plane, FaceHashMap *
         {
             // introduce new vertex if both existing vertices are far from plane
             const auto p = ( o * pd - d * po ) / ( o - d );
-            mesh.splitEdge( e, p, nullptr, new2Old );
+            const auto e1 = mesh.splitEdge( e, p, nullptr, new2Old );
+
+            if ( uvCoords )
+            {
+                const auto org = mesh.topology.org( e1 );
+                const auto dest = mesh.topology.dest( e );
+                if ( org < uvCoords->size() && dest < uvCoords->size() )
```

Comment:
Better check in the beginning that uvCoords have correct size (and avoid check on every subdivision)